course7/homeworks/waleed/ls/main.cc

```diff
@@ -1,16 +1,128 @@
 #include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
 
+#include <string>
+#include <sstream>
 #include <iostream>
 
-int main(int argc, char** argv)
+namespace bpo = boost::program_options;
+namespace bfs = boost::filesystem;
+
+unsigned int fileSize(const bfs::path& path)
 {
-	boost::filesystem::path currPath = boost::filesystem::current_path();
-	boost::filesystem::directory_iterator currDir(currPath);
+	if (!bfs::exists(path))
+		throw std::runtime_error("File doesn't exist");
+
+	if (bfs::is_regular_file(path))
+	{
+		return bfs::file_size(path);
+	}
+	unsigned int result = 0;
+
+	bfs::directory_iterator currDir(path);
 
 	for (const auto& it : currDir)
 	{
-		std::cout << it.path().filename() << std::endl;
+		result += fileSize(it.path());
+	}
+	return result;
+}
+
+std::string treeTraversal(const bfs::path& path, const bool& treeArg, const bool& longArg, unsigned int indentLevel = 0)
```

Comment:
There is no point passing booleans as references. Boolean is a primitive type, one byte, cheap to copy. Cheaper, in fact, than 8 bytes of reference!